File: Copilot/3.Desarrollo Web Endpoint RESTful/golang/endpoint.go
package main

import (
    "fmt"
    "sync"
)

type singleton struct{}

var instance *singleton
var once sync.Once

func GetInstance() *singleton {
    once.Do(func() {
        instance = &singleton{}
    })
    return instance
}

func main() {
    a := GetInstance()
    b := GetInstance()
    fmt.Println("¿Es la misma instancia?", a == b)
}

File: Copilot/3.Desarrollo Web Endpoint RESTful/c++/endpoint.cpp
#include <iostream>
using namespace std;

class Singleton {
private:
    static Singleton* instance;
    Singleton() {}

public:
    static Singleton* getInstance() {
        if (!instance)
            instance = new Singleton();
        return instance;
    }
};

Singleton* Singleton::instance = nullptr;

int main() {
    Singleton* a = Singleton::getInstance();
    Singleton* b = Singleton::getInstance();
    cout << "¿Es la misma instancia? " << (a == b ? "Sí" : "No") << endl;
    return 0;
}
